src/Microsoft.VisualStudio.Composition/Reflection/MemberRef.cs

```diff
@@ -12,7 +12,10 @@ namespace Microsoft.VisualStudio.Composition.Reflection
     using System.Runtime.InteropServices;
     using System.Text;
     using System.Threading.Tasks;
+    using MessagePack;
+    using Microsoft.VisualStudio.Composition.Formatter;
 
+    [MessagePackFormatter(typeof(MemberRefFormatter<MemberRef>))]
```

Comment:
I suspect `UnionAttribute` is a better choice for this base class instead of doing it yourself with a custom formatter.